### PR TITLE
Enhancement to Timeslotbar and Timetable component

### PR DESCRIPTION
**Pull request by choo95dev**
**Changes**
 - add intiial scroll / offset for timetable and time slot bar component

- [delete] branch

**Changelog entry**

- introduce `initialScrollTime` to `timetable` component to set the initial scroll position 
- introduce `initialScrollTime` to `time-slot-bar` component to set the initial scroll position

**Additional information**

- You may refer to this [BOOKINGSG-8534](https://sgtechstack.atlassian.net/browse/BOOKINGSG-8534)

### CODE REVIEW

File: src/util/time-helper.ts [hunk 1]
```diff
@@ -440,11 +458,95 @@ export namespace TimeHelper {
...
+ * Parses a time string in HH:mm format to separate hour and minute values
+ * @param timeString - Time string in HH:mm format (e.g., "09:30", "14:45")
+ * @returns Array containing [hours, minutes] as numbers
+ * @example
+ * parseTimeToNumbers("09:30") // returns [9, 30]
+ * parseTimeToNumbers("14:45") // returns [14, 45]
+ */
+const parseTimeToNumbers = (timeString: string): number[] => {
+    return timeString.split(":").map(Number);
+};
```
Comment (qroll) [inline src/util/time-helper.ts]: nit: can add a new line between functions

File: tests/utils/time-helper.spec.ts [hunk 2]
```diff
@@ -193,3 +193,339 @@ describe("roundToNearestInterval tests", () => {
...
+                minTime: "00:00",
+                maxTime: "12:00",
+                interval: 30,
+                intervalWidth: 60,
+            });
+
+            expect(result).toBe(0);
+        });
+
+        it("should handle end of day (23:59) correctly", () => {
```
[the remark this answers is not in the document]
Comment (choo95dev) [reply tests/utils/time-helper.spec.ts]: removed! , due to it not being a valid edge case

**Review by qroll — APPROVED** [review]
